packages/flutter/test/cupertino/button_test.dart

```diff
@@ -847,6 +847,67 @@ void main() {
     await tester.pump();
     expect(value, isTrue);
   });
+
+  testWidgets('Press and move on button and animation works', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      boilerplate(child: CupertinoButton(onPressed: () {}, child: const Text('Tap me'))),
+    );
+    final TestGesture gesture = await tester.startGesture(
+      tester.getTopLeft(find.byType(CupertinoButton)),
+      kind: switch (defaultTargetPlatform) {
+        TargetPlatform.iOS ||
+        TargetPlatform.android ||
+        TargetPlatform.fuchsia => PointerDeviceKind.touch,
+        TargetPlatform.macOS ||
+        TargetPlatform.windows ||
+        TargetPlatform.linux => PointerDeviceKind.mouse,
+      },
+    );
+    addTearDown(gesture.removePointer);
+    // Check opacity
+    final FadeTransition opacity = tester.widget(
+      find.descendant(of: find.byType(CupertinoButton), matching: find.byType(FadeTransition)),
+    );
+    await tester.pumpAndSettle();
+    final double moveDistance = switch (defaultTargetPlatform) {
```

Comment:
Can you verify that "on mobile, if you move out of the button slightly shorter than `moveDistance`, then the opacity stays the same?